[Bug] Flaky tests in /x/storage/schemaless/projection package

### Bug Description

Test flakiness in x/storage/schemaless/projection package

### Steps to Reproduce

1. Run tests go test -v -count=5 ./x/storage/schemaless/projection
2. Notice that sometimes tests are failing

### Expected Behavior

Regardless of number of runs test are passing

### Actual Behavior

Tests are sometimes passing sometimes are failing.
I not his this behavior in `TestTriggers`
Or it hangs (deadlock?) in `TestPubSubChan`

### Error Output

```shell
=== RUN   TestTriggers
=== RUN   TestTriggers/should_trigger_window_flush_at_watermark
    utils.go:131: 
                Error Trace:    /Users/gabrielhabryn/Work/gh/mkunion/x/storage/schemaless/projection/utils.go:131
                                                        /Users/gabrielhabryn/Work/gh/mkunion/x/storage/schemaless/projection/triggering_test.go:162
                Error:          Not equal: 
                                expected: projection.Item{Key:"key", Data:(*schema.List)(0x1400022f398), EventTime:1546336800100000000, Window:(*projection.Window)(0x1400039a8c0), Type:0x0}
                                actual  : projection.Item{Key:"key", Data:(*schema.List)(0x1400022f698), EventTime:1546336800200000000, Window:(*projection.Window)(0x1400039aa90), Type:0x0}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,17 +3,17 @@
                                  Data: (*schema.List)((len=10) {
                                -  (*schema.Number)(0),
                                -  (*schema.Number)(1),
                                -  (*schema.Number)(2),
                                -  (*schema.Number)(3),
                                -  (*schema.Number)(4),
                                -  (*schema.Number)(5),
                                -  (*schema.Number)(6),
                                -  (*schema.Number)(7),
                                -  (*schema.Number)(8),
                                -  (*schema.Number)(9)
                                +  (*schema.Number)(10),
                                +  (*schema.Number)(11),
                                +  (*schema.Number)(12),
                                +  (*schema.Number)(13),
                                +  (*schema.Number)(14),
                                +  (*schema.Number)(15),
                                +  (*schema.Number)(16),
                                +  (*schema.Number)(17),
                                +  (*schema.Number)(18),
                                +  (*schema.Number)(19)
                                  }),
                                - EventTime: (int64) 1546336800100000000,
                                + EventTime: (int64) 1546336800200000000,
                                  Window: (*projection.Window)({
                                -  Start: (int64) 1546336800000000000,
                                -  End: (int64) 1546336800100000000
                                +  Start: (int64) 1546336800100000000,
                                +  End: (int64) 1546336800200000000
                                  }),
                Test:           TestTriggers/should_trigger_window_flush_at_watermark
=== RUN   TestTriggers/should_trigger_window_emitting_once_at_period_100ms,_and_10_items_arrives_as_1_item
    utils.go:131: 
                Error Trace:    /Users/gabrielhabryn/Work/gh/mkunion/x/storage/schemaless/projection/utils.go:131
                                                        /Users/gabrielhabryn/Work/gh/mkunion/x/storage/schemaless/projection/triggering_test.go:162
                Error:          Not equal: 
                                expected: projection.Item{Key:"key", Data:(*schema.List)(0x1400022f320), EventTime:1546336800100000000, Window:(*projection.Window)(0x1400039a7d0), Type:0x0}
                                actual  : projection.Item{Key:"key", Data:(*schema.List)(0x1400022f890), EventTime:1546336800200000000, Window:(*projection.Window)(0x14000300de0), Type:0x0}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,17 +3,17 @@
                                  Data: (*schema.List)((len=10) {
                                -  (*schema.Number)(0),
                                -  (*schema.Number)(1),
                                -  (*schema.Number)(2),
                                -  (*schema.Number)(3),
                                -  (*schema.Number)(4),
                                -  (*schema.Number)(5),
                                -  (*schema.Number)(6),
                                -  (*schema.Number)(7),
                                -  (*schema.Number)(8),
                                -  (*schema.Number)(9)
                                +  (*schema.Number)(10),
                                +  (*schema.Number)(11),
                                +  (*schema.Number)(12),
                                +  (*schema.Number)(13),
                                +  (*schema.Number)(14),
                                +  (*schema.Number)(15),
                                +  (*schema.Number)(16),
                                +  (*schema.Number)(17),
                                +  (*schema.Number)(18),
                                +  (*schema.Number)(19)
                                  }),
                                - EventTime: (int64) 1546336800100000000,
                                + EventTime: (int64) 1546336800200000000,
                                  Window: (*projection.Window)({
                                -  Start: (int64) 1546336800000000000,
                                -  End: (int64) 1546336800100000000
                                +  Start: (int64) 1546336800100000000,
                                +  End: (int64) 1546336800200000000
                                  }),
                Test:           TestTriggers/should_trigger_window_emitting_once_at_period_100ms,_and_10_items_arrives_as_1_item
    utils.go:131: 
                Error Trace:    /Users/gabrielhabryn/Work/gh/mkunion/x/storage/schemaless/projection/utils.go:131
                                                        /Users/gabrielhabryn/Work/gh/mkunion/x/storage/schemaless/projection/triggering_test.go:162
                Error:          Not equal: 
                                expected: projection.Item{Key:"key", Data:(*schema.List)(0x1400022f338), EventTime:1546336800200000000, Window:(*projection.Window)(0x1400039a7e0), Type:0x0}
                                actual  : projection.Item{Key:"key", Data:(*schema.List)(0x14000312c00), EventTime:1546336800100000000, Window:(*projection.Window)(0x1400039b500), Type:0x0}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -3,17 +3,17 @@
                                  Data: (*schema.List)((len=10) {
                                -  (*schema.Number)(10),
                                -  (*schema.Number)(11),
                                -  (*schema.Number)(12),
                                -  (*schema.Number)(13),
                                -  (*schema.Number)(14),
                                -  (*schema.Number)(15),
                                -  (*schema.Number)(16),
                                -  (*schema.Number)(17),
                                -  (*schema.Number)(18),
                                -  (*schema.Number)(19)
                                +  (*schema.Number)(0),
                                +  (*schema.Number)(1),
                                +  (*schema.Number)(2),
                                +  (*schema.Number)(3),
                                +  (*schema.Number)(4),
                                +  (*schema.Number)(5),
                                +  (*schema.Number)(6),
                                +  (*schema.Number)(7),
                                +  (*schema.Number)(8),
                                +  (*schema.Number)(9)
                                  }),
                                - EventTime: (int64) 1546336800200000000,
                                + EventTime: (int64) 1546336800100000000,
                                  Window: (*projection.Window)({
                                -  Start: (int64) 1546336800100000000,
                                -  End: (int64) 1546336800200000000
                                +  Start: (int64) 1546336800000000000,
                                +  End: (int64) 1546336800100000000
                                  }),
                Test:           TestTriggers/should_trigger_window_emitting_once_at_period_100ms,_and_10_items_arrives_as_1_item
=== RUN   TestTriggers/should_trigger_window_emitting_when_window_size_reach_2_item
--- FAIL: TestTriggers (0.96s)
    --- FAIL: TestTriggers/should_trigger_window_flush_at_watermark (0.32s)
    --- FAIL: TestTriggers/should_trigger_window_emitting_once_at_period_100ms,_and_10_items_arrives_as_1_item (0.32s)
    --- PASS: TestTriggers/should_trigger_window_emitting_when_window_size_reach_2_item (0.32s)


---

=== RUN   TestPubSubChan
WARNING[0008] PubSubChan.Process: no subscribers but get message: 0,"foo" 
(and it stays here, without any progress)
```

### mkunion Version

d7b971aede5ae71ac579d709301c05752801203b

### Go Version

go version go1.24.4 darwin/arm64

### Operating System

macOS

### Code Sample

```go

```

### Impact

High - Significant workaround required

### Workaround

Retry tests until they pass. 

### Checklist

- [x] I have searched for similar issues
- [x] I have provided a minimal reproduction case
- [x] I have included all relevant error messages

--- a/x/storage/schemaless/projection/pub_sub_chan.go
+++ b/x/storage/schemaless/projection/pub_sub_chan.go
@@ -151,7 +151,7 @@ func (s *PubSubChan[T]) Subscribe(f func(T) error) error {
 	newSubscribers := make([]*subscriber[T], 0, len(s.subscribers)-1)
 	for _, su := range s.subscribers {
 		if su != sub {
-			newSubscribers = append(newSubscribers, sub)
+			newSubscribers = append(newSubscribers, su)
 		}
 	}
 	s.subscribers = newSubscribers

